clients/database/database.go

```diff
@@ -0,0 +1,60 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/uptrace/bun"
+	"github.com/uptrace/bun/migrate"
+)
+
+// Migrate sets up and runs all migrations in the migrations model
+// that haven't been run on the database being used by the proxy service
+// returning error (if any) and a list of migrations that have been
+// run and any that were not
+func Migrate(ctx context.Context, db *bun.DB, migrations migrate.Migrations) (*migrate.MigrationSlice, error) {
+	// set up migration config
+	migrator := migrate.NewMigrator(db, &migrations)
+
+	// grab migration lock to prevent race conditions during migration
+	migrator.Lock(ctx)
```

Comment:
i think you need to check the error returned from this to ensure you actually got the lock.

if it comes back non-nil, another process has the lock on the migrations table. maybe sleep and retry in that case?

Reply:
#->TODONE